util/cluster.go

```diff
@@ -0,0 +1,205 @@
+		if matchesValidation(clusterNameOrID, cluster) {
+			clusterIDs = append(clusterIDs, cluster.ID)
+		}
+	}
+
+	if allClusterIDs != nil {
+		CacheClusterIDs(allClusterIDs...)
+	}
+
+	switch {
+	case clusterIDs == nil:
+		return "", microerror.Mask(errors.ClusterNotFoundError)
+
+	case len(clusterIDs) > 1:
+		confirmed, id := handleNameCollision(clusterNameOrID, response.Payload)
+		if !confirmed {
+			return "", nil
+		}
+
+		return id, nil
+
+	default:
+		return clusterIDs[0], nil
+	}
+}
+
+func matchesValidation(nameOrID string, cluster *models.V4ClusterListItem) bool {
+	return cluster.DeleteDate == nil && (cluster.ID == nameOrID || cluster.Name == nameOrID)
+}
+
+func handleNameCollision(nameOrID string, clusters []*models.V4ClusterListItem) (bool, string) {
+	var (
+		clusterIDs []string
+
+		table = []string{color.CyanString("ID | ORGANIZATION | NAME")}
+	)
+
+	for _, cluster := range clusters {
+		if matchesValidation(nameOrID, cluster) {
+			clusterIDs = append(clusterIDs, cluster.ID)
+			table = append(table, fmt.Sprintf("%5s | %5s | %5s\n", cluster.ID, cluster.Owner, cluster.Name))
+		}
+	}
+
+	printNameCollisionTable(table)
+
+	confirmed, id := confirm.AskStrictOneOf(
+		fmt.Sprintf(
+			"Found more than one cluster called '%s', please type the ID of the cluster that you would like to delete",
+			nameOrID,
+		),
+		clusterIDs,
+	)
+	if !confirmed {
+		return false, ""
+	}
+	return true, id
+}
+
+func printNameCollisionTable(table []string) {
+	// Print output
+	fmt.Println("Multiple clusters found")
+	fmt.Printf("\n")
+	fmt.Println(columnize.SimpleFormat(table))
+	fmt.Printf("\n")
+}
+
+// CacheClusterIDs adds cluster IDs to a persistent cache,
+// which can be used for decreasing timeout in getting
+// cluster IDs, for commands that take both cluster names and IDs
+func CacheClusterIDs(c ...string) {
+	fs := config.FileSystem
+
+	existingC := make(chan []string)
+	go func() {
+		e, _ := readClusterCache(fs)
+		existingC <- e
+	}()
+	existing := <-existingC
+
+	var (
+		totalLen    = len(c) + len(existing)
+		allClusters = make([]string, 0, totalLen)
+	)
+
+	allClusters = append(allClusters, existing...)
+	allClusters = append(allClusters, c...)
+	allClusters = removeDuplicates(allClusters)
+
+	writeC := make(chan error)
+	go func() {
+		err := writeClusterCache(fs, allClusters...)
+		writeC <- err
+	}()
+	// Ignore error output
+	_ = <-writeC
+}
+
+// IsInClusterCache checks if a cluster ID is present in the
+// persistent cluster cache
+func IsInClusterCache(ID string) bool {
+	existing, _ := readClusterCache(config.FileSystem)
+
+	for _, name := range existing {
+		if name == ID {
+			return true
+		}
+	}
+
+	return false
+}
+
+func readClusterCache(fs afero.Fs) ([]string, error) {
+	filePath := path.Join(config.ConfigDirPath, clusterCacheFileName)
+	output, err := afero.ReadFile(fs, filePath)
+	if err != nil {
+		return []string{}, err
+	}
+
+	c := strings.Split(string(output), ",")
```

Comment:
The separator should be a constant in the package. BTW, I would prefer a newline. That would make the file much more handy if one should ever have to debug something in a text editor.